### PR TITLE
TINY-11524: Retain user case for CSS property names and color values in Styles parser

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: TINY-11524

Description of Changes:
* Stop lowercasing CSS property names and color values in `Styles.parse`. `color: RED`, `color: var(--MyVar)`, `color: {{MyVar}}`, `Margin-Top: 20px` now round-trip unchanged.
* Internal `behavior:` XSS guard and legacy `font-weight: 700 → bold` substitution route through a local lowercase so mixed-case names still trigger them.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues:
* https://github.com/tinymce/tinymce/issues/7832
* https://github.com/tinymce/tinymce/issues/4204

## Summary by CodeRabbit

* **Bug Fixes**
  * Styles parsing now preserves user-provided casing for custom CSS properties and for many color values in style attributes; standard property names remain normalized and font-weight shorthand still normalizes.
* **Tests**
  * Added/updated tests covering casing preservation, shorthand compression, font-weight normalization, and XSS/behavior guarding.

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/api/html/Styles.ts [hunk 1]
```diff
@@ -255,15 +257,13 @@ const Styles = (settings: StylesSettings = {}, schema?: Schema): Styles => {
...
 
             // Opera will produce 700 instead of bold in their style values
-            if (name === 'font-weight' && value === '700') {
+            if (lowerName === 'font-weight' && value === '700') {
               value = 'bold';
-            } else if (name === 'color' || name === 'background-color') { // Lowercase colors like RED
-              value = value.toLowerCase();
             }
 
             // Convert RGB colors to HEX
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/api/html/Styles.ts:271]: <p class='graphite__hidden'><i>[Re: line +278]</i></p>
I'm not so sure we want to not normalize the property names. Since with this change parsing and expecting the keys to be lowercase would no longer work also you would be able to parse multiple values that define the same thing into the object. For example: `tinymce.html.Styles().parse('font-Weight: bold; font-weight: bold')` or `tinymce.html.Styles().parse('font-Weight: bold')['font-weight']` I think we shoul duse the lowerName name here unless the name starts with `--` because then it's a custom property definition.
<p class='graphite__hidden'>See this comment inline on <a href="https://app.graphite.com/github/pr/tinymce/tinymce/11082?utm_source=unchanged-line-comment">Graphite</a>.</p>

**Review by spocke — CHANGES_REQUESTED** [review]

File: modules/tinymce/src/core/main/ts/api/html/Styles.ts (hunk 1 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/html/Styles.ts:271]: So I didn't realize the implications that this would have if we are to change the case on non custom property things. That could in theory be a breaking change.

Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/api/html/Styles.ts:271]: I agree. I know we've had a lot of complaints about the decision to force lowercase, and Spocke himself logged the ticket to make this change, but I don't think we have enough tests to cover the edge cases this might break.

There are so many places we do style name lookups, are we sure they all use browser APIs where normalisation is done for us?

Comment (ArvinJ-H) [reply modules/tinymce/src/core/main/ts/api/html/Styles.ts:271]: I had a look what downstream consumers and found several lookup sites that assume lowercase keys: `ElementUtils.compare` format merging, schema `valid_styles/invalid_styles` serialize, `paste_webkit_styles` filter, `StyleUtils non-inheritable` set. All breaks under raw case-preserved storage.

We can exception `--`  to address one of the GH issue. would be a one-line fix at `Styles.ts:278` (ie. lower case everything except custom vars.

`styles[name.startsWith('--') ? name : lowerName] = isEncoded ? decode(value, true) : value;`

Original jira ticket does mention 2 GH issues, this closes GH #7832 (`var(--MyVar)`, templating tokens) but not GH #4204 (`Margin-Top:` round-trip for Outlook email).

Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/html/Styles.ts:271]: I think the rule here is that we want to support the HTML and CSS specs and they now allow CSS custom properties to be case sensitive those didn't exist when this code was introduced. The templates strings like `{{fooBar}}` is not standard CSS so not sure we should support that. The browsers just retain the style attribute value as is it doesn't try to normalize it old IE did that is why it was a mess thats why we have the re-boxing of values etc in the serializer.

So I think the safe thing to do here is retain case on the values only and on the keys only if the `--` prefix is there.

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/html/Styles.ts [hunk 2]
```diff
@@ -239,13 +239,17 @@ const Styles = (settings: StylesSettings = {}, schema?: Schema): Styles => {
...
           let value = matches[2].replace(trimRightRegExp, '');
 
           if (name && value) {
             // Decode escaped sequences like \65 -> e
             name = decodeHexSequences(name);
             value = decodeHexSequences(value);
+            // Custom properties (--*) keep user case; standard names normalize to lowercase
+            if (!name.startsWith('--')) {
+              name = name.toLowerCase();
+            }
```
Comment (ArvinJ-H) [inline modules/tinymce/src/core/main/ts/api/html/Styles.ts:252]: @spocke The `toLowerCase` is still happening, its just a moved after custom variable exception check

Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/html/Styles.ts:252]: Yes that fine then we scope it to only things where case really matters and also scoping it to something that is less likely to be used often. 👍

**Review by ltrouton — APPROVED** [review]

**Review by kimwoodfield — APPROVED** [review]
